frontend/src/Metamaps/Listeners.js

```diff
@@ -122,7 +123,24 @@ const Listeners = {
     })
 
     $(window).resize(function () {
-      if (Visualize && Visualize.mGraph) Visualize.mGraph.canvas.resize($(window).width(), $(window).height())
+      if (Visualize && Visualize.mGraph){
+        var canvas = Visualize.mGraph.canvas,
+          scaleX = canvas.scaleOffsetX,
+          scaleY = canvas.scaleOffsetY,
+          centrePixX = Visualize.mGraph.canvas.canvases[0].size.width / 2,
+          centrePixY = Visualize.mGraph.canvas.canvases[0].size.height / 2,
+          centreCoords = Util.pixelsToCoords({x:centrePixX ,y:centrePixY});
+        
+        Visualize.mGraph.canvas.resize($(window).width(), $(window).height())
+  
+        canvas.scale(scaleX,scaleY);
+        var newCentrePixX = Visualize.mGraph.canvas.canvases[0].size.width / 2,
+            newCentrePixY = Visualize.mGraph.canvas.canvases[0].size.height / 2,
+            newCentreCoords = Util.pixelsToCoords({x:newCentrePixX ,y:newCentrePixY});
+        
+        canvas.translate(newCentreCoords.x - centreCoords.x, newCentreCoords.y - centreCoords.y);
```

Comment:
@Bortseb can you describe all of this a little bit? what is the behaviour that all this enables? in particular in terms of centering?